src/commands/epgsql_cmd_start_replication.erl

```diff
@@ -65,8 +65,13 @@ execute(Sock, #start_repl{slot = ReplicationSlot, callback = Callback,
 %% CopyBothResponse
 handle_message(?COPY_BOTH_RESPONSE, _Data, Sock, _State) ->
     {finish, ok, ok, epgsql_sock:set_packet_handler(on_replication, Sock)};
-handle_message(?ERROR, Error, _Sock, _State) ->
+handle_message(?ERROR, Error, Sock, State) ->
+    %% In the case of error, Postgresql replication protocol sends a ReadyForQuery message.
+    %% Adds an error to results to handle it later in the ?READY_FOR_QUERY branch.
     Result = {error, Error},
-    {sync_required, Result};
+    {add_result, Result, Result, Sock, State};
+handle_message(?READY_FOR_QUERY, _Data, Sock, _State) ->
+    [Error = {error, _}] = epgsql_sock:get_results(Sock), % assert a single error response
+    {finish, Error, Error, Sock};
```

Comment:
it should be `{finish, Error, done, Sock}`, because we already sent `{error, _}` from line 72. See how `squery` works for the reference

https://github.com/epgsql/epgsql/blob/bcd1aff3be419dee1c94ef447780d91fad9e8f07/src/commands/epgsql_cmd_squery.erl#L76